src/components/LeafletMap.jsx

```diff
@@ -114,8 +116,9 @@ const LeafletMap = ({ showFilterPane, showMetricsPane, view, setView }) => {
   const _addAndRemoveMarkersBasedOnFilters = useEffect(() => {
     const displayedProjectsChanged = displayedProjects !== prevDisplayedProjects
     const selectedMarkerChanged = selectedMarkerId !== prevSelectedMarkerId
+    const checkedProjectsChanged = checkedProjects !== prevCheckedProjects
```

Comment:
I haven't tried this myself here, but I would assume that if `checkedProjects` is an array this condition would always be `true`. It checks whether the arrays are the same object in memory (which they're not), not whether their contents are identical. 

e.g.
```js
const array1 = [1, 2, 3]
const array2 = [1, 2, 3]

console.log(array1 === array2) // false
```

Consider writing a function or using a utility function to do a deep compare.

Reply:
It seems to be doing a deep compare. I tried toggling one of the checkboxes and this is what I see in my console. If nothing changes then it is false, but when the checkboxes is toggled then it changes to true
![image](https://github.com/user-attachments/assets/e688513c-f034-4350-865a-f5483d58d7fc)